python/cudf/cudf/options.py

```diff
@@ -114,6 +115,40 @@ def set_option(name: str, val: Any):
     option.value = val
 
 
+class option_context(ContextDecorator):
+    """
+    Context manager to temporarily set options in the `with` statement context.
+
+    You need to invoke as ``option_context(pat, val, [(pat, val), ...])``.
+
+    Examples
+    --------
+    >>> from cudf import option_context
+    >>> with option_context('display.max_rows', 10, 'display.max_columns', 5):
+    ...     pass
+    """
+
+    def __init__(self, *args) -> None:
+        if len(args) % 2 != 0 or len(args) < 2:
+            raise ValueError(
```

Comment:
Can we add tests of `cudf.option_context` itself? e.g. set options, get options, ensure that the "undo" works, that errors are raised properly, etc.

Reply:
Done 👍